apps/dailykanji/dailykanji.star

```diff
@@ -9,156 +9,237 @@ load("cache.star", "cache")
+            render.Marquee(width = int(SCREEN_WIDTH / 2), child = render.Text(row_text, color = text_colors[i], font = FONT)),
+            left = int(SCREEN_WIDTH / 2),
+            top = top_margin + i * (FONT_HEIGHT + V_SPACING),
+        ))
+
+    scroll_delay = int(config.get("scroll", "45"))
     return render.Root(
-        render.Stack(
-            children = display_items,
-        ),
+        render.Stack(children = display_items),
         show_full_animation = True,
+        delay = scroll_delay // 2 if canvas.is2x() else scroll_delay,
     )
 
-def add_padding_to_child_element(element, left = 0, top = 0, right = 0, bottom = 0):
-    padded_element = render.Padding(
-        pad = (left, top, right, bottom),
-        child = element,
-    )
-    return padded_element
-
-def get_random_kanji(KANJI_IMAGE_LIST):
-    kanji_images = KANJI_IMAGE_LIST.split(" ")
-    random_number = random(0, len(kanji_images))
-    i = 0
-    for item in kanji_images:
-        if random_number == i:
-            return item
-        i = i + 1
-
-    # Should never get here.
-    return kanji_images[0]
-
-def random(min, max):
-    now = time.now()
-    rand = int(str(now.nanosecond)[-6:-3]) / 1000
-    return int(rand * (max - min) + min)
-
-def display_kanji_with_image_url(individual_kanji, kanji_alive_key):
-    i = 0
-    for kanjicharacter in individual_kanji:
-        i = i + 1
-        url = KANJI_API_URL % kanjicharacter
-        kanji_http = http.get(url)
-        kanji_data = kanji_http.json()
-        print(i)
-        kanji_image_json = get_kanji_information(kanji_data["kanji"], kanji_alive_key)
-        if "kanji" not in kanji_image_json:
-            print("%s %s" % (kanjicharacter, i))
-
-#unused for now, but if we want to refactor later and get the image
-#and convert, we could do that instead of storing all that base64 stuff in the file here
-def get_kanji_image(image_url):
-    return http.get(image_url).body()
-
-#Using the kanjialive api to get information on a particular kanji character
-def get_kanji_information(selected_kanji, kanji_alive_key):
-    res = http.get(
-        url = KANJI_ALIVE_URL % selected_kanji,
-        headers = {
-            "X-RapidAPI-Host": "kanjialive-api.p.rapidapi.com",
-            "X-RapidAPI-Key": kanji_alive_key,
-        },
-    )
-
-    if res.status_code == 200:
-        return res.json()
-    else:
-        print("Error")
-        return None
-
 def get_schema():
+    scroll_speed_options = [schema.Option(display = d, value = v) for d, v in [("Slow Scroll", "60"), ("Medium Scroll", "45"), ("Fast Scroll", "30")]]
+
     return schema.Schema(
         version = "1",
         fields = [
             schema.Text(
                 id = "api_key",
                 name = "API Key",
-                desc = "Create a rapidapi account and go to https://rapidapi.com/KanjiAlive/api/learn-to-read-and-write-japanese-kanji/playground for key",
+                desc = "RapidAPI key for KanjiAlive (optional)",
```

Comment:
Could you please retain the URL for the API signup somewhere? Could be here, in the manifest `desc` attribute, or in a README. I've seen other apps that just say "RapidAPI key" and it's hard for users to figure out how to get one (this one is slightly better because it names the API).